documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -93,225 +66,84 @@ include::config_examples/serialization_marshaller.xml[]
+=== JBoss Marshalling
+JBoss Marshalling is a Serialization based marshalling library which was the default marshaller in past versions of {brandname}.
+It's possible to still utilise this library, however it's necessary to first add the `infinipsan-jboss-marshalling` dependency
+to the classpath. No additional configuration is required, as if this module is on the class path then the user marshaller defaults
+to using JBoss Marshalling.
 
-----
-java.io.NotSerializableException: java.lang.Object
-at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:857)
-at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:407)
-at org.infinispan.marshall.exts.ReplicableCommandExternalizer.writeObject(ReplicableCommandExternalizer.java:54)
-at org.infinispan.marshall.jboss.ConstantObjectTable$ExternalizerAdapter.writeObject(ConstantObjectTable.java:267)
-at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:143)
-at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:407)
-at org.infinispan.marshall.jboss.JBossMarshaller.objectToObjectStream(JBossMarshaller.java:167)
-at org.infinispan.marshall.VersionAwareMarshaller.objectToBuffer(VersionAwareMarshaller.java:92)
-at org.infinispan.marshall.VersionAwareMarshaller.objectToByteBuffer(VersionAwareMarshaller.java:170)
-at org.infinispan.marshall.DefaultMarshallerTest.testNestedNonSerializable(VersionAwareMarshallerTest.java:415)
-Caused by: an exception which occurred:
-in object java.lang.Object@b40ec4
-in object org.infinispan.commands.write.PutKeyValueCommand@df661da7
-... Removed 22 stack frames
-----
-
-The way the "in object" messages are read is the same in which stacktraces are read. The highest "in object" being the most inner one and the lowest "in object" message being the most outer one. So, the above example indicates that a java.lang.Object instance contained in an instance of org.infinispan.commands.write.PutKeyValueCommand could not be serialized because java.lang.Object@b40ec4 is not serializable.
-
-This is not all though! If you enable DEBUG or TRACE logging levels, marshalling exceptions will contain show the toString() representations of objects in the stacktrace. For example:
-
-----
-java.io.NotSerializableException: java.lang.Object
-...
-Caused by: an exception which occurred:
-in object java.lang.Object@b40ec4
--> toString = java.lang.Object@b40ec4
-in object org.infinispan.commands.write.PutKeyValueCommand@df661da7
--> toString = PutKeyValueCommand{key=k, value=java.lang.Object@b40ec4, putIfAbsent=false, lifespanMillis=0, maxIdleTimeMillis=0}
-----
+WARNING: link:{javadocroot}/org/infinispan/commons/marshall/AdvancedExternalizer.html[AdvancedExternalizer] implementations
```

Comment:
FYI admonitions should use this format:

/[TIP/NOTE/IMPORTANT/WARNING] /
/====
admonition text
====/

Reply:
/[WARNING] /
/====
{brandname} ignores implementations of the link:{javadocroot}/org/infinispan/commons/marshall/AdvancedExternalizer.html[AdvancedExternalizer] interface when persisting data unless you configure JBoss marshalling. However, this interface is also deprecated and planned for removal.
====/

Reply:
Done